### ISSUE TITLE
ValueError: non-hexadecimal number found in fromhex() arg

### ISSUE THREAD

**Issue by jwigert** [EMPI-PL/SWOS_career_team_explorer#11]
When opening a `.CAR` file, the program crashes with the following error:

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.11/tkinter/__init__.py", line 1967, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "/home/kali/dosgames/SensiWs9/SWOS_career_team_explorer/CARexplorer.py", line 42, in opencar
    carfile_output = readcarfile(pickedfile)  # Call func to read squad details
                     ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kali/dosgames/SensiWs9/SWOS_career_team_explorer/carread.py", line 99, in readcarfile
    hex2ascii(hexcutzeros(hexpure(hexread(open_infile, call+3, 23)))),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kali/dosgames/SensiWs9/SWOS_career_team_explorer/func.py", line 31, in hex2ascii
    tmpdata = bytes.fromhex(hexdata).decode(code)
          ^^^^^^^^^^^^^^^^^^^^^^
```

To analyze the issue, I rewrote function `hex2ascii`:

```py
def hex2ascii(hexdata, code='utf-8'):
    # Converts hex data into ascii
    try:
        tmpdata = bytes.fromhex(hexdata).decode(code)    
        return tmpdata
    except ValueError as e:
        print(f"Error decoding hexdata: {hexdata}")
        print(str(e))
```

When I executed the program again, I got the following log:

```
Error decoding hexdata: 4a4f455920424541554348414d5
non-hexadecimal number found in fromhex() arg at position 27
```

The hexdata with an additional trailing 0 translates to the player name JOEY BEAUCHAMP.

Do you have any idea as to why the error occurs? If not, would you be willing to accept a PR adding the exception handling, so that the career file can be opened without the program crashing?

**Comment by Jezza53** [comment]
The issue is to do with the last letter of that player's name (just had the same issue with player "AO DIOP"). 

Letter "P" in hex is "50" (and full name is '414f2044494f**50**00524500000000000000000000000000' - with trailing zeros).

So when the code runs:
`hex2ascii(hexcutzeros(hexpure(hexread(open_infile, call+3, 23))))`
when initialising the player, it "eats up" the last (valid) zero from the hex value, in effect attempting to run:
`hex2ascii('414f2044494f5')`
or
`bytes.fromhex('414f2044494f5')`
which throws an error, as not a valid hex value (unlike '414f2044494f50', which is - and translates to "AO DIOP").

I fixed the issue (quick fix) by amending the hexcutzeros function as:

```
def hexcutzeros(hexdata):
    # Cut off entire string post initial 00 position
    # NOTE: This will avoid misspelled names in squad
    tmpdata = hexdata[:hexdata.index('00')] 
    #to avoid error if last letter is "P" (i.e. "50" in hex) and hexcutzeros has truncated the last "0"
    if len(tmpdata) % 2 != 0:
        tmpdata += '0'  
    return tmpdata
```

Basically adding a trailing zero when length of hexdata is not an even number (assuming a last zero had been incorrectly eaten up). 

I've just submitted a PR for this change.

PS: Great tool! Love it :) Thanks a lot for creating it.

Jezza
